plugins/MauticFocusBundle/Event/FocusViewEvent.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * @copyright   2017 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace MauticPlugin\MauticFocusBundle\Event;
+
+use MauticPlugin\MauticFocusBundle\Entity\Focus;
+use MauticPlugin\MauticFocusBundle\Entity\Stat;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Class FocusViewEvent.
+ */
+class FocusViewEvent extends Event
+{
+    /**
+     * @var Focus
+     */
+    private $focus;
+
+    /**
+     * @var Stat
+     */
+    private $stat;
+
+    /**
+     * @param Stat $stat
+     */
+    public function __construct(Stat $stat)
+    {
+        $this->stat  = $stat;
+        $this->focus = $stat->getFocus();
```

Comment:
This seems redundant to me. The listeners can do this as well. You could extract all props from the stat entity like Lead. Let's remove this and keep the event simple.

Reply:
I would like use Stat $stat in event. 
In my plugin I need work with stat id and date_added to stat. 
Than what you recommend? Just remove focus property? Or add another props to event , somesthin like

public function __construct(Stat $stat, Focus $focus, Lead $contact)

?

Reply:
I'd leave only the `$stat` property in the `FocusViewEvent` class. In your listener you can 
```
$stat = $event->getStat();
$focus = $stat->getFocus();
$contact = $stat->getLead();
// ... and so on
```
Setting the `$focus` property to the event from the `$stat` entity seems like redundant code to me as you can do the same thing in the listener.

Reply:
Now I got it. 
Thanks